rust/ql/lib/codeql/rust/elements/internal/FormatArgumentImpl.qll

```diff
@@ -40,6 +40,16 @@ module Impl {
 
     override Format getParent() { result = Synth::TFormat(parent, index, _, _) }
 
+    /** Gets the position of this argument. */
+    int getPosition() {
+      this =
+        rank[result + 1](FormatArgument f, int offs |
+          f = Synth::TFormatArgument(parent, index, _, _, _, offs)
+        |
+          f order by offs
+        )
+    }
```

Comment:
I think the following would be simpler. However `getPosition()` is a bit meaningless. We could rename it to `getKind()`, but having a numeric kind value does not seem very nice either. So I still think this should be removed or made private.

```suggestion
    int getPosition() {
      this = Synth::TFormatArgument(parent, index, result, _, _, _)        
    }
```